include/deal.II/lac/relaxation_block.templates.h

```diff
@@ -151,6 +153,39 @@ RelaxationBlock<MatrixType,inverse_type>::invert_diagblocks ()
   this->inverses_computed(true);
 }
 
+namespace internal
+{
+  /**
+   * default implementation for serial vectors. Here we don't need to make a
+   * copy into a ghosted vector, so just return a reference to @p prev.
+   */
+  template <class VECTOR, class VEC1>
+  const VECTOR &
+  prepare_ghost_vector(
+    const VECTOR &prev,
+    const VEC1 * /*other*/)
+  {
+    return prev;
+  }
+
+  /**
+   * Specialization for Trilinos. Use the ghosted vector.
+   */
+  template <>
+  const TrilinosWrappers::MPI::Vector &
+  prepare_ghost_vector(
+    const TrilinosWrappers::MPI::Vector &prev,
+    const TrilinosWrappers::MPI::Vector *other)
+  {
+    Assert(other!=NULL,
+           ExcMessage("You need to provide a ghosted vector in RelaxationBlock::AdditionalData::temp_trilinos_ghost_vector."));
+    Assert(other->size()==prev.size(), ExcInternalError());
+
+    // import ghost values:
+    *const_cast<TrilinosWrappers::MPI::Vector *>(other) = prev;
+    return *other;
```

Comment:
Do we have a policy on thread-safety? Do we mark thread-safe or non-thread-safe code? This is function is called from a ``vmult`` of a preconditioner, so I don't think you want to call those concurrently.